src/VisualStudio/CSharp/Impl/CSharpVSResources.resx

```diff
@@ -540,6 +540,57 @@
   <data name="Event_Hookup_Command_Handler" xml:space="preserve">
     <value>Generate Event Subscription Command Handler</value>
   </data>
+  <data name="Remove_unnecessary_usings" xml:space="preserve">
+    <value>Remove unnecessary usings</value>
+  </data>
+  <data name="Sort_usings" xml:space="preserve">
+    <value>Sort usings</value>
+  </data>
+  <data name="Format_document_settings" xml:space="preserve">
+    <value>Format Document Settings (Experiment) </value>
+  </data>
+  <data name="All_csharp_formatting_rules" xml:space="preserve">
+    <value>All C# Formatting Rules (indentation, wrapping and spacing)</value>
+  </data>
+  <data name="Fix_accessibility_modifiers" xml:space="preserve">
+    <value>Fix accessibility modifiers</value>
+  </data>
+  <data name="Fix_add_remove_braces" xml:space="preserve">
+    <value>Fix add/remove braces</value>
+  </data>
+  <data name="Fix_framework_types" xml:space="preserve">
+    <value>Fix framework types (int vs Int32)</value>
+  </data>
+  <data name="Fix_implicit_explicit_type" xml:space="preserve">
+    <value>Fix implicit/explicit type</value>
+  </data>
+  <data name="Fix_this_qualification" xml:space="preserve">
+    <value>Fix this. qualification</value>
+  </data>
+  <data name="Sort_accessibility_modifiers" xml:space="preserve">
+    <value>Sort accessibility modifiers</value>
+  </data>
+  <data name="Fix_expression_bodied_members" xml:space="preserve">
+    <value>Fix expression-bodied members</value>
+  </data>
+  <data name="Fix_inline_variable_declarations" xml:space="preserve">
+    <value>Fix inline variable declarations</value>
+  </data>
+  <data name="Fix_language_features" xml:space="preserve">
+    <value>Fix language features</value>
+  </data>
+  <data name="Fix_object_collection_initialization" xml:space="preserve">
+    <value>Fix object/collection initialization</value>
+  </data>
+  <data name="Make_readonly" xml:space="preserve">
+    <value>Make readonly</value>
+  </data>
+  <data name="Remove_unnecessary_casts" xml:space="preserve">
+    <value>Remove unnecessary casts</value>
+  </data>
+  <data name="Remove_unused_variables" xml:space="preserve">
+    <value>Remove unused variables</value>
+  </data>
```

Comment:
needs options for the parentheses styles that went in.

Reply:
That seems really unfortunate.  It feels like this is then a hodge-podge of fixes, instead of being a natural and complimentary way to ensure that all the existing cod-style stuff can naturally just be integrated into commands like "format" (or "save").  

It seems like it would be *much* better if this was an actual system that analyzers/fixers could opt-into.  Then, when features (like 'parentheses code styles') were written, they could just opt into (or maybe out-of) this system and it would all light up. 

The current approach seems like an incredible amount of hard-coded knowledge and fixed functions were used instead of a simpler and less knowledgeable system that would then require far less work for all these features.